atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -42,15 +42,38 @@ private const val MIN_SUPPORTED_KGP_VERSION = "1.7.0"
 open class AtomicFUGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) = project.run {
         checkCompatibility()
-        val pluginVersion = rootProject.buildscript.configurations.findByName("classpath")
-            ?.allDependencies?.find { it.name == "atomicfu-gradle-plugin" }?.version
+        // Get the version of atomicfu-gradle-plugin applied to the project:
+        // first try to find it in the root buildscript, then in the buildscript of the current project.
+        // The error is thrown in case atomicfu-gradle-plugin is not applied.
+        val pluginVersion = rootProject.getAGPVersion() ?: this.getAGPVersion()
+        requireNotNull(pluginVersion) { "We were unable to find the version of `kotlinx-atomicfu` plugin applied to the project.\n" +
+                "Please ensure that `kotlinx-atomicfu` plugin is correctly applied.\n" +
+                "Below is an example of Kotlin DSL code that applies `kotlinx-atomicfu` plugin:\n" +
+                "```\n" +
+                "buildscript {\n" +
+                "    repositories {\n" +
+                "        mavenCentral()\n" +
+                "    }\n" +
+                "\n" +
+                "    dependencies {\n" +
+                "      classpath(\"org.jetbrains.kotlinx:atomicfu-gradle-plugin:0.23.2\")\n" +
+                "    }\n" +
+                "}\n" +
+                "\n" +
+                "apply(plugin = \"kotlinx-atomicfu\")\n" +
+                "```\n" +
+                "For further information, refer to the project's README: https://github.com/Kotlin/kotlinx-atomicfu/blob/master/README.md#apply-plugin"}
         extensions.add(EXTENSION_NAME, AtomicFUPluginExtension(pluginVersion))
         applyAtomicfuCompilerPlugin()
         configureDependencies()
         configureTasks()
     }
 }
 
+private fun Project.getAGPVersion(): String? =
```

Comment:
Sorry to say that, but the AGP abbreviation was already taken by the Android Gradle Plugin, so it worth using some other name here. 

It took me a while to understand that the problem you fixed has nothing to do with android. :)

Reply:
Sorry, fixed that 🥲